pandas/core/arrays/timedeltas.py

```diff
@@ -316,14 +314,6 @@ def __mul__(self, other):
                 freq = self.freq * other
             return type(self)(result, freq=freq)
 
-        if not hasattr(other, "dtype"):
-            # list, tuple
-            other = np.array(other)
-        if len(other) != len(self) and not is_timedelta64_dtype(other):
-            # Exclude timedelta64 here so we correctly raise TypeError
-            #  for that instead of ValueError
-            raise ValueError("Cannot multiply with unequal lengths")
```

Comment:
This is a small change (reflected in a test below).  `tdarr * tdarr[:-1]` checks for length mismatch before checking for dtype compat, so now raises ValueError instead of TypeError.